### PR TITLE
TI: avoid uint64×uint64 ClockP time math; extend tick wrap

### PR DESCRIPTION
**Pull request by hcm444**
See commit message; addresses TI timestamp review from #166.

### CODE REVIEW

File: firmware/common/utils/ti_hal_time.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+uint32_t ti_get_time_since_boot_ms(void) {
+    const uint32_t period_us = ClockP_getSystemTickPeriod();
+    const uint64_t ticks = ti_extended_system_ticks_u64();
+    // 1000 µs/tick ⇒ 1 ms per tick: elapsed ms matches tick count (uint32 wraps with API).
+    if (period_us == 1000U) {
+        return static_cast<uint32_t>(ticks);
+    }
+    const uint64_t us = ticks * static_cast<uint64_t>(period_us);
```
Comment (CoolNamesAllTaken) [inline firmware/common/utils/ti_hal_time.cpp]: This still includes a 64 bit multiply that will be called every time we're looking for a timestamp on the TI MCU.

**Comment by hcm444** [comment]
> This is starting to look too complicated. I think this could really just be a one or two line code change with a static uint32_t to count the number of wraps in the existing function.

To turn extended tick count into microseconds you still need something like tick_count * period_us (unless period_us == 1, or you special-case e.g. 1000 with shifts)

also restoring my previous commit logic here, but I gotta wait a LONG time to see if it passes tests so give me some time

**Comment by hcm444** [comment]
Yeah I asked ChatGPT for help with the helper methods for edge cases around 64 bit multiplication on the CC1312 (which I don't have on me) it should still work.

**Comment by CoolNamesAllTaken** [comment]
At this point all of the issues mentioned in this PR should be fixed by other changes, or have fixes in progress.
